routes/actions.js

```diff
@@ -347,4 +347,12 @@ function sendRegistrationForm(req, res, formInfo) {
   });
 }
 
+// checks for external redirect in returnTo, ignores if present and redirects
+function returnToPath(req, res) {
+    let returnTo = req.query.returnTo;
+    if (returnTo.includes('//')) {
```

Comment:
That test again won't be as reliable as the regex. Try, for example:

http://localhost/signin?returnTo=ftp:rtfm.mit.edu
http://localhost/signin?returnTo=mailto:some@address.examle (with mailto handler configured)

Reply:
OK, I misunderstood your example.  We're trying to only allow paths that start with / since these will always be relative.  //, protocol:// and protocol: are all valid which means testing for them would be a fools errand and break often.

I'll implement the regexp.